indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -86,6 +89,11 @@
  */
 public class TaskLockbox
 {
+  private static final long LOCK_ACQUIRE_TIMEOUT_SECONDS = 10L;
+
+  private final Map<String, ReadWriteLock> datasourceToConcurrentLock = new HashMap<>();
+  private final Map<String, Integer> datasourceToNumActiveLocks = new HashMap<>();
```

Comment:
Where is this count used?

Reply:
Thanks for catching this. 
The concurrent lock needs to be removed from the first map when removing entries from this map.
Do you think this is unneeded?

Reply:
Yes, we can remove this.

Reply:
Done